pkg/kine/server/create.go

```diff
@@ -28,13 +36,18 @@ func (l *LimitedServer) create(ctx context.Context, put *etcdserverpb.PutRequest
 		return nil, unsupported("prevKv")
 	}
 
+	createCnt.Add(ctx, 1)
+
 	rev, err := l.backend.Create(ctx, string(put.Key), put.Value, put.Lease)
+	span.SetAttributes(attribute.Int64("revision", rev))
 	if err == ErrKeyExists {
+		span.RecordError(err)
 		return &etcdserverpb.TxnResponse{
 			Header:    txnHeader(rev),
 			Succeeded: false,
 		}, nil
 	} else if err != nil {
+		span.RecordError(err)
```

Comment:
Can we use defer for this?

Reply:
So `span.RecordError(err)` does not stop the span, it will also record just `nil`. Why would you like to use defer here?